File: src/app/Skill.java
/*
 * Decompiled with CFR 0.152.
 */
package app;

import java.io.InputStream;

class Skill {
    public static int ski_id;
    public int you;
    public int param;
    public String name;

    Skill(int n, InputStream inputStream, InputStream inputStream2) {
        byte[] byArray;
        ski_id = n;
        try {
            byArray = new byte[4];
            inputStream.read(byArray);
            this.you = byArray[3] & 0xFF;
            this.param = (byArray[0] & 0xFF) << 8 | byArray[1] & 0xFF;
        }
        catch (Exception exception) {
            // empty catch block
        }
        try {
            int n2;
            byArray = new byte[16];
            inputStream2.read(byArray);
            for (n2 = 0; n2 < 16 && byArray[n2] != 0; ++n2) {
            }
            byte[] byArray2 = new byte[n2];
            for (int i = 0; i < n2; ++i) {
                byArray2[i] = byArray[i];
            }
            this.name = new String(byArray2);
        }
        catch (Exception exception) {
            // empty catch block
        }
    }
}

File: src/app/Skill.cpp
#include "Skill.h"

int Skill::ski_id = 0;

Skill::Skill(int n, InputStream *inputStream, InputStream *inputStream2)
    : you(0), param(0)
{
    ski_id = n;
    if (inputStream) {
        try {
            unsigned char byArray[4];
            inputStream->read(byArray, 4);
            you   = byArray[3] & 0xFF;
            param = ((byArray[0] & 0xFF) << 8) | (byArray[1] & 0xFF);
        } catch (...) {}
    }
    if (inputStream2) {
        try {
            unsigned char byArray[16];
            inputStream2->read(byArray, 16);
            int n2 = 0;
            while (n2 < 16 && byArray[n2] != 0) n2++;
            name = sjis_to_utf8(byArray, n2);
        } catch (...) {}
    }
}
